src/documents/tests/test_consumer.py

```diff
@@ -17,6 +30,33 @@ def test__get_parser_class_1_parser(self):
             self.DummyParser
         )
 
+    def test_open_errors(self):
+        """
+        Test that exceptions on open do not crash the consumer
+
+        It is not enough to test the consumption process with `open` throwing
+        an error everytime.  The first open call could succeed, but then the
+        file gets moved or permission changed by another process (e.g. user
+        manually intervening or a second paperless process).  Thus we need to
+        check what happens when only the n-th call fails.
+        """
+        
+        # open can throw PermissionError or FileNotFoundError
+        for error in [PermissionError(), FileNotFoundError()]:
+            # Check exception on n-th open call
+            for n in range(0, 4):
```

Comment:
I'm not sure what test runner paperless uses but there should be a way to have those two `for` loops be parameters.
See [pytest documentation](https://docs.pytest.org/en/latest/parametrize.html) for what I mean.

Reply:
As far as I understand currently the tests are written in the `unittest.TestCase` style and pytest [does not support parametrization of tests written in that style](https://docs.pytest.org/en/latest/unittest.html#pytest-features-in-unittest-testcase-subclasses). Is pytest "given" and can I write tests (with parameterization) that will only work with pytest but e.g. not the "standard library unittest" module?

Reply:
Do you mean "We don't want to use PyTest style"? Because pytest is currently a [dependency](https://github.com/the-paperless-project/paperless/blob/master/requirements.txt#L54) of paperless. I'm not sure where/how tests are actually run, but I think it's also done [by pytest (by travis)](https://github.com/the-paperless-project/paperless/blob/master/.travis.yml#L37).